### ISSUE TITLE
SPM fails to start with customized cache location

### ISSUE THREAD

**Issue by gmacon** [saltstack/salt#46451]
### Description of Issue/Question

I've been using the workaround I posted in [a comment on #36479](https://github.com/saltstack/salt/issues/36479#issuecomment-303836693) to build SPM packages as a non-root user. As new developers have joined our team, they've noticed that the build fails for them with this error:

```
Failed to create directory path "/var/cache/salt/master" - [Errno 13] Permission denied: '/var/cache/salt'
```

### Setup

You need [this script](https://gist.github.com/gmacon/ab201fbe6a3b2a40664042dae62df6ab) containing the workaround and a formula to build. Any formula will do.

### Steps to Reproduce Issue

```
$ ./spm_build.py --log-level=debug --log-file=./spm.log --verbose build foo-formula
[DEBUG   ] Reading configuration from /tmp/tmpBuI7qJ/etc/spm
[DEBUG   ] Reading configuration from /tmp/tmpBuI7qJ/etc/spm
[DEBUG   ] Missing configuration file: ~/.config/salt/saltrc
[DEBUG   ] Configuration file path: /tmp/tmpBuI7qJ/etc/spm
Failed to create directory path "/var/cache/salt/master" - [Errno 13] Permission denied: '/var/cache'
Traceback (most recent call last):
  File "./spm_build.py", line 37, in <module>
    call_spm(sys.argv[1:], 'repo')
  File "./spm_build.py", line 31, in call_spm
    args)
  File "/usr/local/var/pyenv/versions/2.7.14/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['spm', '-c', '/tmp/tmpBuI7qJ/etc', '--log-level=debug', '--log-file=./spm.log', '--verbose', 'build', 'foo-formula']' returned non-zero exit status 13
$ file spm.log
spm.log: empty
```

### Versions Report

```
Salt Version:
           Salt: 2017.7.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.5.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

Note that the build environment doesn't have the salt master installed, so I ran `salt-call --versions-report` instead.

### Bisect

I bisected this error to 58378866e53bac9379a66ad1f485d5e6739f3374.

I don't see anything obviously wrong with that commit.

**Comment by gtmanfred** [comment]
Can you apply this diff to `salt/cli/spm.py` and see if it fixes your problem?

```
diff --git a/salt/cli/spm.py b/salt/cli/spm.py
index 3cecc76905..b279cecc2a 100644
--- a/salt/cli/spm.py
+++ b/salt/cli/spm.py
@@ -30,7 +30,7 @@ class SPM(parsers.SPMParser):
         self.parse_args()
         self.setup_logfile_logger()
         v_dirs = [
-            self.config['cachedir'],
+            self.config['spm_cache_dir'],
         ]
         verify_env(v_dirs,
                    self.config['user'],
```

Thanks,
Daniel

**Comment by gmacon** [comment]
That fixed it. I guess it was an obvious problem :-)

**Comment by gtmanfred** [comment]
Yeah, i didn't know enough about what I was actually fixing in the commit i guess.

I will submit a pr and get this in 2017.7.5 hopefully

**Comment by gmacon** [comment]
OK, I confirm this is fixed in 2017.7.5. What's the procedure for handling a forward-port to 2018.3?

**Comment by gtmanfred** [comment]
It will be in 2018.3.1
